implementations/rust/ockam/ockam/src/entity/profile.rs

```diff
@@ -31,40 +69,258 @@ impl Profile {
         identifier: ProfileIdentifier,
         change_history: ProfileChangeHistory,
         verification_policies: Vec<ProfileVerificationPolicy>,
+        vault: Arc<Mutex<dyn ProfileVault>>,
     ) -> Self {
         Profile {
             identifier,
             change_history,
             verification_policies,
+            vault,
         }
     }
 }
 
 impl Profile {
-    pub fn apply(&mut self, change_event: ProfileChangeEvent) {
-        if !self.verify(&change_event) {
-            return; // TODO: Throw error
+    pub fn create_with_key(
+        label: String,
+        key_type: ProfileKeyType,
```

Comment:
can we collapse key_type and key_purpose into the key_attributes?

Reply:
@mikelodder7 definitely. Do you think label should also go there?

Reply:
yes, I think it should. It's just another attribute of the key.